sqlstatements.go

```diff
@@ -41,44 +44,80 @@ func getSQLMetrics(s Sql_server, q Query) (metrics []graphite.Metric, err error)
 	}
 	defer db.Close()
 
-	rows, err := db.Query(q.Tsql)
-	if err != nil {
-		log.Println(err)
-		return nil, err
-	}
-	defer rows.Close()
+	// FIXME: there should be a startegy
+	if len(q.Tsql_row) != 0 {
 
-	columnNames, err := rows.Columns()
-	if err != nil {
-		log.Println(err)
-		return nil, err
-	}
+		rows, err := db.Query(q.Tsql_row)
+		if err != nil {
+			return nil, err
+		}
+		defer rows.Close()
 
-	values := make([]interface{}, len(columnNames))
-	valuePointers := make([]interface{}, len(columnNames))
-	for i := 0; i < len(columnNames); i++ {
-		valuePointers[i] = &values[i]
-	}
+		columnNames, err := rows.Columns()
+		if err != nil {
+			return nil, err
+		}
 
-	rows.Next()
-	if err = rows.Scan(valuePointers...); err != nil {
-		log.Println(err)
-		return nil, err
-	}
+		values := make([]interface{}, len(columnNames))
+		valuePointers := make([]interface{}, len(columnNames))
+		for i := 0; i < len(columnNames); i++ {
+			valuePointers[i] = &values[i]
+		}
 
-	if err = rows.Err(); err != nil {
-		log.Println(err)
-		return nil, err
-	}
+		rows.Next()
+		if err = rows.Scan(valuePointers...); err != nil {
+			return nil, err
+		}
 
-	for i := 0; i < len(columnNames); i++ {
-		metrics = append(metrics, graphite.Metric{
-			fmt.Sprintf("%s.%s", s.Metric_prefix, NormalizeMetricName(columnNames[i])),
-			fmt.Sprintf("%v", values[i]),
-			time.Now().Unix(),
-		})
-	}
+		if err = rows.Err(); err != nil {
+			return nil, err
+		}
+
+		for i := 0; i < len(columnNames); i++ {
+			metrics = append(metrics, graphite.Metric{
+				fmt.Sprintf("%s.%s", q.Metric_prefix, NormalizeMetricName(columnNames[i])),
+				fmt.Sprintf("%v", values[i]),
+				time.Now().Unix(),
+			})
+		}
+
+		return metrics, nil
+
+	} else if len(q.Tsql_table) != 0 {
```

Comment:
Looks like if one gives both `tsql_table` and `tsql_row` only the latter has effect and no exception is thrown.
Let's add some validation to config that one cannot supply both of them -- or allow both of them to work as one would expect.